app/lib/repository/client.dart

```diff
@@ -11,3 +14,45 @@ Future<Client> login(String username, String password) async {
   Client client = await sdk.login(username, password);
   return client;
 }
+
+//Fetch user id from sdk
+Future<String> getUser(Future<Client> client) async {
+  Client _client = await client;
+  final String userId = await _client.userId();
+  return userId;
+}
+
+//Fetch messages from sdk and store them in list
+Future<List<types.Message>> getMessages(
+  TimelineStream stream,
+  int count,
+  Conversation room,
+) async {
+  List<types.Message> _messages = [];
+  bool isSeen = false;
+  var messages = await stream.paginateBackwards(count);
+  for (RoomMessage message in messages) {
+    //Based on boolean, it'll update the status of message (seen,delivered) etc
+    await room.readReceipt(message.eventId()).then(
+          (value) => {
+            isSeen = value,
+          },
+        );
+    types.TextMessage m = types.TextMessage(
+      id: message.eventId(),
+      showStatus: true,
+      author: types.User(id: message.sender()),
+      text: message.body(),
+      status: isSeen ? Status.seen : Status.delivered,
+    );
+    _messages.add(m);
+    isSeen = !isSeen;
+  }
+  return _messages;
+}
+
+//Push message
+Future<String> sendMessage(Conversation convo, String message) async {
+  var res = await convo.sendPlainMessage(message);
+  return res;
+}
```

Comment:
Same here? What is the purpose of a function doing exactly one function call and returning its result? Why?